ttnn/cpp/ttnn/operations/data_movement/common/common.hpp

```diff
@@ -156,6 +156,65 @@ ttnn::Tensor pad_to_tile_vol(
     const bool use_multicore,
     const std::optional<MemoryConfig>& memory_config);
 
+enum class ShardStrategy { BLOCK, HEIGHT, WIDTH };
+
+ttnn::MemoryConfig create_sharded_memory_config(
+    const ttnn::Shape& shape,
+    const tt::tt_metal::CoreRangeSet& core_grid,
+    const ShardStrategy& strategy,
+    const tt::tt_metal::ShardOrientation& orientation,
+    bool halo = false,
+    bool use_height_and_width_as_shard_shape = false,
+    const tt::tt_metal::Layout& layout = tt::tt_metal::Layout::ROW_MAJOR);
+
+template <ShardStrategy strategy>
+std::pair<uint32_t, std::vector<uint32_t>> tensor_coord_to_shard_coord(
+    const std::span<const uint32_t>& tensor_shape,
+    const std::span<const uint32_t>& shard_shape,
+    const std::span<const uint32_t>& tensor_coord) {
+    static_assert(strategy == ShardStrategy::HEIGHT, "FIXME(jkruer): Only height strategy is supported currently.");
```

Comment:
Let's rename this function to indicate this limitation. We can rename it back once the limitations are lifted.

Reply:
Done